cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModel.java

```diff
@@ -1391,4 +1409,16 @@ private void refreshCapacity() {
       _clusterCapacity[r.id()] = capacity;
     }
   }
+
+  private boolean sanityCheckPartitionInfoConsistency(PartitionInfo partitionInfo) {
+    Partition partition = partition(new TopicPartition(partitionInfo.topic(), partitionInfo.partition()));
+    if (partition == null) {
+      return false;
+    }
+
+    Set<Integer> replicasFromPartition = partition.replicas().stream().map(Replica::broker).map(Broker::id).collect(Collectors.toSet());
+    Set<Integer> replicasFromPartitionInfo = Arrays.stream(partitionInfo.replicas()).map(Node::id).collect(Collectors.toSet());
+    return partition.leader().broker().id() == partitionInfo.leader().id()
```

Comment:
Are there any side-effects if we omit the leader equality check?

Reply:
If we omit the check, the real leader replica can be deleted. That case it may introduce extra leadership movement, which I think might be fine.